rust/connlib/tunnel/src/client.rs

```diff
@@ -503,6 +507,95 @@ impl ClientState {
         Some(packet)
     }
 
+    pub(crate) fn handle_dns_response(&mut self, response: dns::RecursiveResponse) {
```

Comment:
The whole dns thing I think is  getting quite big in a module that's already huge. Do you mind extracting this to a separate `client_dns` module? Feel free to change the name

Reply:
Agreed. I made a note in the project board about that already! Basically, the entire `dns` module is client-side only, apart from perhaps a few utility functions so all of it should probably move into the `client` directory.